Netflix Clone/Services/Encoder.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  Encoder.swift
+//  Netflix Clone
+//
+//  Created by Polina Poznyak on 6.07.23.
+//
+
+import Foundation
+
+public struct Encoder {
+    static func encodeParameters(for urlRequest: inout URLRequest, with parameters: [String: Any]) throws {
+        guard let url = urlRequest.url else { throw NetworkError.missingURL }
+
+        if var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: false), !parameters.isEmpty {
+            urlComponents.queryItems = [URLQueryItem]()
+            for (key,value) in parameters {
+                let queryItem = URLQueryItem(name: key, value: "\(value)")
+                urlComponents.queryItems?.append(queryItem)
+            }
+            urlRequest.url = urlComponents.url
+        }
+    }
+
+    static func setHeaders(for urlRequest: inout URLRequest, with headers: [String: String]) throws {
+        for (key, value) in headers{
+            urlRequest.setValue(value, forHTTPHeaderField: key)
+        }
+    }
+}
```

Comment:
Nice 👍 

On iOS,  however, it makes more sense to have an extension to URLRequest
So the API will be
`urlRequest.encode(parameters: [String: Any]`
vs
`Encoder.encodeParameters(for request, with parameters)`

And same for the header